libraries/Microsoft.Bot.Builder/ICredentialTokenProvider.cs

```diff
@@ -33,6 +33,19 @@ public interface ICredentialTokenProvider : IUserTokenProvider
         /// <remarks>If the task completes successfully, the result contains the raw signin link.</remarks>
         Task<string> GetOauthSignInLinkAsync(ITurnContext turnContext, AppCredentials oAuthAppCredentials, string connectionName, CancellationToken cancellationToken);
 
+        /// <summary>
+        /// Get the raw signin link to be sent to the user for signin for a connection name, using customized AppCredentials.
+        /// </summary>
+        /// <param name="turnContext">Context for the current turn of conversation with the user.</param>
+        /// <param name="oAuthAppCredentials">AppCredentials for OAuth.</param>
+        /// <param name="connectionName">Name of the auth connection to use.</param>
+        /// <param name="emulatorUrl">Url of the emulator service.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        /// <remarks>If the task completes successfully, the result contains the raw signin link.</remarks>
+        Task<string> GetOauthSignInLinkAsync(ITurnContext turnContext, AppCredentials oAuthAppCredentials, string connectionName, string emulatorUrl, CancellationToken cancellationToken);
```

Comment:
>GetOauthSignInLinkAsync [](start = 21, length = 23)

this is a breaking change, can't do it.

Reply:
yes good call. i'll implement a new interface for this method.